test/e2e/main_test.go

```diff
@@ -3,61 +3,149 @@ package e2e
+	cleanupScriptPath := flag.String("cleanup-script-path", "../../hack/run-test-cleanup.sh", "The absolute/relative path to the testing cleanup hack script.")
+	testOutputPath := flag.String("test-output-path", "", "The absolute/relative path that you want to store test logs within.")
+	reportingAPIURL := flag.String("reporting-api-url", "", "reporting-operator URL if useKubeProxyForReportingAPI is false")
 	httpsAPI := flag.Bool("https-api", false, "If true, use https to talk to Metering API")
 	useKubeProxyForReportingAPI := flag.Bool("use-kube-proxy-for-reporting-api", false, "If true, uses kubernetes API proxy to access reportingAPI")
 	useRouteForReportingAPI := flag.Bool("use-route-for-reporting-api", true, "If true, uses a route to access reportingAPI")
-	reportingAPIURL := flag.String("reporting-api-url", "", "reporting-operator URL if useKubeProxyForReportingAPI is false")
-	routeBearerToken := flag.String("route-bearer-token", "", "The bearer token to the reporting-operator serviceaccount if use-route-for-reporting-api is set to true")
+	logLevel := flag.String("log-level", logrus.DebugLevel.String(), "The log level")
 	flag.Parse()
 
-	var err error
-	if testReportingFramework, err = reportingframework.New(*ns, *kubeconfig, *httpsAPI, *useKubeProxyForReportingAPI, *useRouteForReportingAPI, *routeBearerToken, *reportingAPIURL, reportTestOutputDirectory); err != nil {
-		logrus.Fatalf("failed to setup framework: %v\n", err)
+	logger := testhelpers.SetupLogger(*logLevel)
+
+	if testOutputDirectory == "" {
+		testOutputDirectory = *testOutputPath
+	}
+
+	loggingPath, err := ioutil.TempDir(testOutputDirectory, *nsPrefix)
+	if err != nil {
+		logger.Fatalf("Failed to create the directory '%s' to log test output: %v", testOutputDirectory, err)
+	}
+
+	logger.Infof("Logging resource and container logs to '%s'", loggingPath)
+
+	cfg := deployframework.ReportingFrameworkConfig{
+		HTTPSAPI:                    *httpsAPI,
+		UseKubeProxyForReportingAPI: *useKubeProxyForReportingAPI,
+		UseRouteForReportingAPI:     *useRouteForReportingAPI,
+		ReportingAPIURL:             *reportingAPIURL,
+		KubeConfigPath:              *kubeConfigFlag,
+	}
+
+	if df, err = deployframework.New(cfg, logger, *nsPrefix, *manifestDir, *cleanupScriptPath, loggingPath); err != nil {
+		logger.Fatalf("Failed to create a new deploy framework: %v", err)
 	}
 
 	os.Exit(m.Run())
 }
 
-func TestReportingProducesData(t *testing.T) {
+func TestMultipleInstalls(t *testing.T) {
+	defaultTargetPods := 7
+	defaultPlatform := "openshift"
+
+	testInstallConfigs := []struct {
+		TargetPods int
+		Name       string
+		Config     deploy.Config
+	}{
+		{
+			Name:       "hdfsInstall",
+			TargetPods: defaultTargetPods,
+			Config: deploy.Config{
+				Platform:        defaultPlatform,
+				DeleteNamespace: true,
+				MeteringConfig: &meteringv1.MeteringConfig{
+					Spec: testhelpers.NewMeteringConfigSpec(),
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testInstallConfigs {
```

Comment:
after this line:
```
t  := t
testCase := testCase
```
https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables